media/client/main/source/MediaPipeline.cpp

```diff
@@ -134,15 +136,16 @@ MediaPipelineFactory::createMediaPipeline(std::weak_ptr<IMediaPipelineClient> cl
     {
         std::shared_ptr<client::IMediaPipelineIpcFactory> mediaPipelineIpcFactoryLocked = mediaPipelineIpcFactory.lock();
         std::shared_ptr<client::IClientController> clientControllerLocked = clientController.lock();
-        mediaPipeline = std::make_unique<client::MediaPipeline>(client, videoRequirements,
-                                                                mediaPipelineIpcFactoryLocked
-                                                                    ? mediaPipelineIpcFactoryLocked
-                                                                    : client::IMediaPipelineIpcFactory::getFactory(),
-                                                                common::IMediaFrameWriterFactory::getFactory(),
-                                                                clientControllerLocked
-                                                                    ? *clientControllerLocked
-                                                                    : client::IClientControllerAccessor::instance()
-                                                                          .getClientController());
+        firebolt::rialto::client::IClientController &cc =
+            clientControllerLocked ? *clientControllerLocked
+                                   : client::IClientControllerAccessor::instance().getClientController();
+
+        auto mp{std::make_shared<client::MediaPipeline>(client, videoRequirements,
+                                                        mediaPipelineIpcFactoryLocked
+                                                            ? mediaPipelineIpcFactoryLocked
```

Comment:
What about a new registerclient api that's called from constructor?

Reply:
I discussed that with Phil earlier. The problem is that unregister client is called in destructor and (with enable_shared_from_this), shared_from_this doesn't work in destructor (makes sense, it's a virtual function). I suggested to call unregister client in shared_ptr custom deleter, but it also throws an exception